flang/include/flang/Lower/Support/BoxValue.h

```diff
@@ -209,6 +212,112 @@ class BoxValue : public AbstractBox, public AbstractArrayBox {
 /// Used for triple notation (array slices)
 using RangeBoxValue = std::tuple<mlir::Value, mlir::Value, mlir::Value>;
 
+/// Set of variables (addresses) holding the allocatable properties. These may
+/// be empty in case it is not deemed safe to duplicate the descriptor
+/// information locally (e.g, volatile entities).
```

Comment:
The wording of the comment may be bad. The point is not to say "empty mutable value implies volatile" it is more to say: "If we need a single point of truth for the mutable values, then we use the descriptor and this set of variables will be empty. We want this single point of truth aspect for volatile entities for instance".

Reply:
Ah. Yeah, it would be good to rephrase this comment then.

We could be assertive here:
```c++
/// A volatile allocatable will be always be lowered to a descriptor to preserve the integrity of the entity and its associated properties. As such, all references to the entity and its property will go through the descriptor explicitly.
```

Reply:
Thanks, I used the rephrasing.